src/Explorer/Tree/Collection.ts

```diff
@@ -690,14 +690,23 @@ export default class Collection implements ViewModels.Collection {
   }
 
   public onNewMongoShellClick() {
-    const id = this.container.tabsManager.getTabs(ViewModels.CollectionTabKind.MongoShell).length + 1;
+    const mongoShellTabs = this.container.tabsManager.getTabs(
+      ViewModels.CollectionTabKind.MongoShell
+    ) as MongoShellTab[];
+
+    let index = 1;
+    if (mongoShellTabs.length > 0) {
+      index = mongoShellTabs[mongoShellTabs.length - 1].getIndex() + 1;
```

Comment:
>getIndex [](http://example.com/codeflow?start=56&length=8)

i like this index being part of the tab..

but just curious: would just an incrementing index as the count in tabsManager would not just work?

Reply:
discussed offline. Synced with Deborah as well. She liked the experience in my PR as opposed to keeping an overall tab count for each type and incrememnting it, irrespective of what was the last opened tab.